app/models/criterion.rb

```diff
@@ -154,6 +154,13 @@ def scale_marks
     unless updated_marks.empty?
       Mark.upsert_all(all_marks.pluck_to_hash.map { |h| { **h.symbolize_keys, mark: updated_marks[h['id'].to_i] } })
     end
+
+    AnnotationCategory.where(flexible_criterion_id: self.id).each do |category|
```

Comment:
It's not enough to just check `id` here; guard this by checking whether `self` is a `FlexibleCriterion`

Reply:
(In which case you can do `self.annotation_categories`)